### DISCUSSION TITLE
Questions about submitting

### DISCUSSION THREAD

**Discussion by ForestsKing** [PaddlePaddle/PaddleSpatial#65] [BaiduKDDCup2022] [answered]
When I submit my code, I get the error:'NoneType' object has no attribute 'forecast'.
My local folder structure:
pytorch
|- checkpoints
|- predict.py
|- prepare.py
|- ...
Then I packaged the pytorch folder into a zip and uploaded it, where is the error?

**Comment by ramber1836** [comment, marked as answer]
The real error is that "No module named 'pytorch' "
Your predict.py is mainly based on the developed utilities in the folder,
namely pytorch.
However, the predict.py is located in the folder where no folder named
'pytorch' was found, so your predict.py cannot find the module of pytorch.
I guess your files in the submitted zip file (e.g. pytorch.zip) should
follow the directory structure like this
   pytorch/
   | --- pytorch/
             | --- checkpoints/
             | --- ...
   | --- predict.py
   | --- prepare.py
   | --- ...

**Reply by ForestsKing** [reply]
Sorry, I don't quite understand what you mean. Why do I need to build another pytorch in the pytorch folder and put the checkpoints folder in it？

**Reply by ramber1836** [reply]
Let me put this another way, 

In your code of the predict.py, you have 

from pytorch.exp import EXP

While in the evaluation.py, the path to the predict.py is something like this: 

/tmp/xxx/pytorch/predict.py

The path of /tmp/xxx/pytorch is added to the system path, and no folders named pytorch can be found under the newly added path. 

So, the thing is evaluation.py does not know how to import pytorch given the newly added path.

In other words, the root folder name (i.e. 'pytorch', in your case) in your submitted zip file does not really matter, you can name it whatever you want. 
The structure under the root directory that how your scripts and model folder are organized is the thing you should really care about.

**Reply by ForestsKing** [reply]
Then why can't I use relative import .exp import EXP ? Thank you very much

**Comment by ramber1836** [comment]
This is a common issue when programming in Python, please refer to this
thread
https://stackoverflow.com/questions/16981921/relative-imports-in-python-3

**Comment by mb-Ma** [comment]
NoneType' object has no attribute 'forecast'.
This error should be caused by missing __init__.py
